cli/src/complete.rs

```diff
@@ -557,13 +613,8 @@ pub fn branch_name_equals_any_revision(current: &std::ffi::OsStr) -> Vec<Complet
         // Don't complete branch names since we want to create a new branch
         return Vec::new();
     };
-    all_revisions()
+    revisions(None, revision)
```

Comment:
Perhaps, we can use `revset_expression()` here. The right-hand side is a revset expression.

Reply:
`revset_expression()` takes an `&OsStr` as expression, and we already converted to `&str` to perform the split, so I'm not sure how we could do this.

(By the way, another thing is bothering me now: I just noticed that `revisions()` and `revset_expression()` take the same arguments, apart from string type, but the arguments are in different orders!)

Reply:
I wasn't aware of `AsRef<OsStr>`, fixed now